server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/DesiredBalanceResponse.java

```diff
@@ -56,26 +60,30 @@ public void writeTo(StreamOutput out) throws IOException {
     }
 
     @Override
-    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject();
-        {
+    public Iterator<? extends ToXContent> toXContentChunked() {
+        Iterator<ToXContent> header = Iterators.single((builder, params) -> {
+            builder.startObject();
             builder.startObject("stats");
             stats.toXContent(builder, params);
             builder.endObject();
-        }
-        {
             builder.startObject("routing_table");
-            for (Map.Entry<String, Map<Integer, DesiredShards>> indexEntry : routingTable.entrySet()) {
-                builder.startObject(indexEntry.getKey());
-                for (Map.Entry<Integer, DesiredShards> shardEntry : indexEntry.getValue().entrySet()) {
-                    builder.field(String.valueOf(shardEntry.getKey()));
-                    shardEntry.getValue().toXContent(builder, params);
-                }
-                builder.endObject();
+            return builder;
+        });
+        Iterator<ToXContent> routingTableIterator = routingTable.entrySet().stream().map(indexEntry -> (ToXContent) (builder, params) -> {
+            builder.startObject(indexEntry.getKey());
+            for (Map.Entry<Integer, DesiredShards> shardEntry : indexEntry.getValue().entrySet()) {
+                builder.field(String.valueOf(shardEntry.getKey()));
+                shardEntry.getValue().toXContent(builder, params);
             }
             builder.endObject();
-        }
-        return builder.endObject();
+            return builder;
+        }).iterator();
+        Iterator<ToXContent> footer = Iterators.single((builder, params) -> {
+            builder.endObject();
+            builder.endObject();
+            return builder;
+        });
+        return Iterators.concat(header, routingTableIterator, footer);
```

Comment:
I have a slight preference to inline these local variables, just to avoid any risk of these things being declared out-of-order.